app/src/main/java/com/sopt/now/presentation/auth/login/LoginActivity.kt

```diff
@@ -0,0 +1,75 @@
+package com.sopt.now.presentation.auth.login
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import com.sopt.now.presentation.main.MainActivity
+import com.sopt.now.databinding.ActivityLoginBinding
+import com.sopt.now.model.login.RequestLoginDto
+import com.sopt.now.presentation.auth.signup.SignupActivity
+import com.sopt.now.utils.Constants.Companion.MEMBER_ID
+import com.sopt.now.utils.NetworkUtil
+import java.lang.Exception
+
+class LoginActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityLoginBinding
+    private val loginViewModel by viewModels<LoginViewModel>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityLoginBinding.inflate(layoutInflater).apply { setContentView(root) }
+
+        with(binding) {
+            btnSignup.setOnClickListener { navigateToSignup() }
+            btnLogin.setOnClickListener { login() }
+        }
+
+        observeLogin()
+    }
+
+    private fun login() {
+        loginViewModel.postLogin(getLoginData())
+    }
+
+    private fun observeLogin() {
+        loginViewModel.status.observe(this) {
+            if (it) {
+                val memberId = loginViewModel.getMemberId()
+                navigateToMain(memberId)
+                showToast(memberId.toString())
+            } else {
+                showToast(loginViewModel.getErrorMessage() ?: "")
+            }
+        }
+    }
+
+    private fun getLoginData(): RequestLoginDto {
+        with(binding) {
+            val id = etLoginId.text.toString().trim()
+            val password = etLoginPw.text.toString().trim()
+
+            return RequestLoginDto(
+                authenticationId = id,
+                password = password
+            )
+        }
+    }
+
+    private fun navigateToSignup() =
+        Intent(this, SignupActivity::class.java).apply { startActivity(this) }
+
+    private fun navigateToMain(userId: String?) {
+        Intent(this, MainActivity::class.java).apply {
+            putExtra(MEMBER_ID, userId)
+            startActivity(this)
+            finish()
+        }
+    }
```

Comment:
scope함수의 사용을 정확하게 알고 쓰는 것도 좋을 것 같아요!

apply의 경우 초기화를 담당하는데 putExtra함수를 사용하는 것은 적합하지만 함수 본문을 실행하는 startActivity와 finish에서의 사용은 적합하지 않다 생각합니다! 객체 초기화를 한 후 객체 자신(this)를 반환하기 때문에 객체에 여러 설정을 적용할 때 적합하다 합니다.

also의 경우 객체의 초기화 또는 설정을 수행한 후, 해당 객체에 대한 추가 작업을 수행하고자 할때 사용합니다. also는 호출된 객체(it을 통해 접근)를 그대로 반환하기 때문에, 객체의 초기화나 설정 이후에 연쇄적인 작업을 할 때 유용하게 사용된다고 합니다!

```suggestion
    private fun navigateToMain(userId: String?) {
        Intent(this, MainActivity::class.java).apply {
           putExtra(MEMBER_ID, userId)
        }.also {
           startActivity(it)
        }
        finish()
    }
```
혹은
```kotlin
private fun navigateToMain(userId: String?) {
    val intent = Intent(this, MainActivity::class.java).apply {
        putExtra(MEMBER_ID, userId)
    }
    startActivity(intent)
    finish()
}
```

Reply:
오오 scope 함수 다시 공부해야겠네요! 감사합니다!